Add openGraph metadata [skip percy]

CharlyMartin opened the pull request:
## 📝 Description

This PR adds open graph and Twitter metadata to FFDW.

## 🛠️ Key Changes

- Renames `metaTitle` and `metaDescription` to `title` and `description`
- Use `{ absolute: SEO.title }` instead of `overrideTitle`
- Clean: Rename `structuredDataParams` to `StructuredDataParams`

## 🧪 How to Test

https://vercel.com/filecoin-foundations-projects/ffdweb-site/7LcgXssGpXy84AR2en3x3MXBrPCF/og

## 📸 Screenshots

![CleanShot 2025-03-17 at 15 06 49@2x](https://github.com/user-attachments/assets/3513b59d-5a4d-4922-beda-19bcbfba5123)

mirhamasala commented on apps/ffdweb-site/src/app/_utils/generateWebPageStructuredData.ts:
```diff
@@ -1,14 +1,12 @@
 import { type DynamicPathValues, type PathValues } from '@/constants/paths'
 import { BASE_URL } from '@/constants/siteMetadata'
 
 type GenerateWebPageStructuredDataProps = {
```
Can be extended from `StructuredDataParams`

mirhamasala commented on apps/ffdweb-site/src/app/_utils/createMetadata.ts:
```diff
@@ -1,23 +1,62 @@
+  const { type = 'website' } = openGraph
+  const { card = 'summary_large_image' } = twitter
+
+  const images = [image || graphicsData.imageFallback.data.src]
```
Fallback should be `homepage`

mirhamasala commented on apps/ffdweb-site/src/app/privacy-policy/page.tsx:
```diff
@@ -10,25 +10,27 @@ import { MarkdownContent } from '@/components/MarkdownContent'
 }
 
 export const metadata = createMetadata({
+  title: { absolute: PRIVACY_POLICY_SEO.title },
```
Shouldn't be overridden.

mirhamasala commented on apps/ffdweb-site/src/app/terms-of-use/page.tsx:
```diff
@@ -10,25 +10,27 @@ import { MarkdownContent } from '@/components/MarkdownContent'
 }
 
 export const metadata = createMetadata({
+  title: { absolute: TERMS_OF_USE_SEO.title },
```
Not absolute

mirhamasala commented on apps/ffdweb-site/src/app/_utils/createMetadata.ts:
```diff
@@ -1,23 +1,62 @@
+  twitter = {},
+}: MetadataParams) {
+  const { type = 'website' } = openGraph
+  const { card = 'summary_large_image' } = twitter
```
Should be `summary`

mirhamasala reviewed (changes requested):
@CharlyMartin - Left some comments that need addressing.

Also, is passing open-graph data through the CMS for collections accounted for?

CharlyMartin commented on apps/ffdweb-site/src/app/blog/[slug]/page.tsx:
```diff
@@ -64,12 +64,15 @@ export async function generateStaticParams() {
+    description,
     path: `${PATHS.BLOG.path}/${slug}`,
-    overrideTitle: true,
+    image: image?.src,
```
https://filecoinfoundation.slack.com/archives/C02LUK9FXK2/p1742236867429669

mirhamasala replied:
Added a ticket here [>>](https://www.notion.so/filecoin/FFDW-Blog-Include-fallback-image-for-Open-Graph-1ba7631f2825809b8115dc31790cc0cd?pvs=4)

CharlyMartin commented:
> Also, is passing open-graph data through the CMS for collections accounted for?

@mirhamasala I reworked the metadata and added support for CMS meta config, which was not implemented. Thanks for flagging!

As I was implementing it, I noticed some things that didn't make so much sense to me:
- `open_graph_config` has `title`, `description` and `image`, but `twitter_config` does not. Do you know why?
- Also, is there a world where opengraph's & twitter's title/description need to differ from the page meta title and description?

mirhamasala reviewed (approved):
@CharlyMartin ~ Code is looking good to me, so let's get this merged! ✨ @barbaraperic - I'm going to open a separate ticket to test the implementation with different scenarios. 🙏🏼
